pywbem_mock/_wbemconnection_mock.py

```diff
@@ -3043,31 +3043,34 @@ def _fake_invokemethod(self, methodname, objectname, Params, **params):
                              local_only=False, include_qualifiers=True,
                              include_classorigin=True)
 
+        # Determine if method defined in classname defined in
+        # the classorigin of the method
         try:
-            target_class = cc.methods[methodname].class_origin
+            target_cln = cc.methods[methodname].class_origin
         except KeyError:
             raise CIMError(CIM_ERR_METHOD_NOT_FOUND, 'Method %s not found '
-                           'in %s class hiearchy' % (methodname,
-                                                     localobject.classname))
-
-        # test for target class in methods repo
+                           'in class %s class' % (methodname,
+                                                  localobject.classname))
+        if target_cln != cc.classname:
+            # TODO FUTURE: add method to repo that allows privileged users
+            # direct access so we don't have to go through _get_class and can
+            # test classes directly in repo
+            tcc = self._get_class(target_cln, namespace,
+                                  local_only=False, include_qualifiers=True,
+                                  include_classorigin=True)
+            if methodname not in tcc.methods:
+                raise CIMError(CIM_ERR_METHOD_NOT_FOUND, 'Method %s not found '
+                               'in class %s' % (methodname, target_cln))
```

Comment:
I think it would be useful for understanding this situation that the exception message mentions that this is in the origin class of the method.

Reply:
I agree and changed the messages that involve the origin class.

DONE